tests/console/zypper_lr.pm

```diff
@@ -12,11 +12,117 @@ use base "consoletest";
+    assert_script_run "zypper lr -d";
+    save_screenshot;
+    script_run "clear";
+
+    my $base_product = "SLES";
+
+    # Server & Desktop DVD
+    if (check_var('FLAVOR', 'Server-DVD')) {
+        validatelr({product => "SLES", uri => "cd:///"});
+    }
+    elsif (check_var('FLAVOR', 'SAP-DVD')) {
+        validatelr({product => "SLE-", product_channel => "SAP", uri => "cd:///"});
+    }
+    elsif (check_var('FLAVOR', 'Server-DVD-HA')) {
+        validatelr(
+            {
+                product => "SLES",
+                uri     => "cd:///"
+            });
+        validatelr(
+            {
+                product => 'SLE-*HA',
+                uri     => get_var('ADDONURL_HA') || "dvd:///"
+            });
+        if (exists $h_addonurl{geo} || exists $h_addons{geo}) {
+            validatelr(
+                {
+                    product => 'SLE-*HAGEO',
+                    uri     => get_var('ADDONURL_GEO') || "dvd:///"
+                });
+        }
+        delete @h_addonurl{"ha", "geo"};
+        delete @h_addons{"ha",   "geo"};
+    }
+    elsif (check_var('FLAVOR', 'Desktop-DVD')) {
+        validatelr({product => "SLED", uri => "cd:///"});
+        $base_product = "SLED";
+    }
+    elsif (is_jeos) {
+        validatelr({product => "SLES", product_channel => "Pool",    uri => "https://updates.suse.com"});
+        validatelr({product => "SLES", product_channel => "Updates", uri => "https://updates.suse.com"});
+    }
+
+    # URI Addons
+    for my $addonurl (keys %h_addonurl) {
+        validatelr({product => $addonurl, uri => get_var("ADDONURL_$addonurl")});
+    }
+
+    # DVD Addons
+    for my $addon (keys %h_addons) {
+        validatelr({product => $addon, uri => "dvd:///"});
+    }
+
+    # SCC Addons: verify existence of 'Pool' and 'Updates' online sources from SCC for
+    # base SLES/SLES and addons
+    if (check_var('SCC_REGISTER', 'installation')) {
+        validatelr(
+            {
+                product         => $base_product,
+                product_channel => check_var('SCC_REGISTER', 'installation') ? "Pool" : undef,
+                uri             => "https://.*suse"
+            });
+        validatelr(
+            {
+                product         => $base_product,
+                product_channel => check_var('SCC_REGISTER', 'installation') ? "Updates" : undef,
+                uri             => "https://.*suse"
+            });
+    }
+    for my $sccaddon (keys %h_scc_addons) {
+        validatelr(
+            {
+                product         => $sccaddon,
+                product_channel => check_var('SCC_REGISTER', 'installation') ? "Pool" : undef,
+                uri             => "https://.*suse"
+            });
+        validatelr(
+            {
+                product         => $sccaddon,
+                product_channel => check_var('SCC_REGISTER', 'installation') ? "Updates" : undef,
+                uri             => "https://.*suse"
+            });
+    }
 }
```

Comment:
why not merge the two blocks for `$base_product` and `%h_scc_addons`? You can put both `$base_product` and `keys %h_scc_addons` in the same list and iterate over both